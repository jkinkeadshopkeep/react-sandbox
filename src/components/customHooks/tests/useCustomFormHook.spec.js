import TestComponent from "./testComponents/formTestComponent";
import {render, screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('#useCustomFormHook', () => {

    describe('when initial form values initialized', () => {
        it('should render input values correctly', () => {
            const {asFragment} = render(<TestComponent/>);
            screen.debug();
            expect(asFragment()).toMatchSnapshot();
        });
    });

    describe('when input value is updated', () => {
        it('update and render correctly', () => {
            const {asFragment, getByLabelText} = render(<TestComponent/>);
            const input = getByLabelText('Forename');
            userEvent.clear(input);
            userEvent.type(input, "Bob");
            expect(input.value).toBe('Bob');
            expect(asFragment()).toMatchSnapshot();
        });
    });

    describe('when form is cleared', () => {
        it('update and render correctly', () => {
            const {asFragment, getByText, getByLabelText} = render(<TestComponent/>);
            const clearButton = getByText('Clear Index');
            const forenameInput = getByLabelText('Forename');
            const surnameInput = getByLabelText('Surname');

            userEvent.click(clearButton);
            expect(forenameInput.value).toBe('');
            expect(surnameInput.value).toBe('');
            expect(asFragment()).toMatchSnapshot();
        });
    });
});