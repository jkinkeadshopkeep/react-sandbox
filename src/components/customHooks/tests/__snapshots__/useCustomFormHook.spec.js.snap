// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#useCustomFormHook when form is cleared update and render correctly 1`] = `
<DocumentFragment>
  <form>
    <label
      for="forename"
    >
      Forename
    </label>
    <input
      id="forename"
      name="forename"
      type="text"
      value=""
    />
    <label
      for="surname"
    >
      Surname
    </label>
    <input
      id="surname"
      name="surname"
      type="text"
      value=""
    />
    <button
      type="button"
    >
      Clear Form
    </button>
  </form>
</DocumentFragment>
`;

exports[`#useCustomFormHook when initial form values initialized should render input values correctly 1`] = `
<DocumentFragment>
  <form>
    <label
      for="forename"
    >
      Forename
    </label>
    <input
      data-testid="forename"
      id="forename"
      name="forename"
      type="text"
      value="Joe"
    />
    <label
      for="surname"
    >
      Surname
    </label>
    <input
      data-testid="surname"
      id="surname"
      name="surname"
      type="text"
      value="Bloggs"
    />
    <button
      type="button"
    >
      Clear Form
    </button>
  </form>
</DocumentFragment>
`;

exports[`#useCustomFormHook when input value is updated update and render correctly 1`] = `
<DocumentFragment>
  <form>
    <label
      for="forename"
    >
      Forename
    </label>
    <input
      id="forename"
      name="forename"
      type="text"
      value="Bob"
    />
    <label
      for="surname"
    >
      Surname
    </label>
    <input
      id="surname"
      name="surname"
      type="text"
      value="Bloggs"
    />
    <button
      type="button"
    >
      Clear Form
    </button>
  </form>
</DocumentFragment>
`;
